analysis/data_utils.py

```diff
@@ -42,6 +39,65 @@ def drop_uninteresting_columns(experiment_df):
     ]]
 
 
+FuzzerBenchmark = collections.namedtuple('FuzzerBenchmark',
+                                         ['fuzzer', 'benchmark', 'experiment'])
+
+
+def clobber_snapshots(experiment_df, experiments):
+    """Clobber snapshots that are part of lower priority (generally earlier)
+    versions of the same trials in |experiment_df|. For example in experiment-1
+    we may test fuzzer-a on benchmark-1. In experiment-2 we may again test
+    fuzzer-a on benchmark-1 because fuzzer-a was updated. This function will
+    remove the snapshots from fuzzer-a,benchmark-1,experiment-1 from
+    |experiment_df| because we want the report to only contain the up-to-date
+    data. Experiment priority is determined by order of each experiment in
+    |experiments| with the highest priority experiment coming last in that
+    list."""
+    # Create a set of tuples that contain all the unique
```

Comment:
Would it be simpler to build up the clobbered table, eg like this?

```python
    # Include everything from the last experiment.
    experiments.reverse()
    last_exp_name = experiments[0]
    result = experiment_df[experiment_df.experiment == last_exp_name]
    for exp_name in experiments[1:]:
        # Include data for not yet covered benchmark/fuzzer pairs.
        covered_pairs = result[['benchmark', 'fuzzer']].drop_duplicates()
        covered_pairs = covered_pairs.apply(tuple, axis=1).tolist()
        exp = experiment_df[experiment_df.experiment == exp_name]
        exp_pairs = exp[['benchmark', 'fuzzer']].apply(tuple, axis=1)
        to_include = exp[~exp_pairs.isin(covered_pairs)]
        result = result.append(to_include)
    return result

```

Reply:
Yeah much simpler actually. Thanks!

Reply:
Any opinion on whether I should reverse the order of the `result` so it more resembles the initial order?